### PR TITLE
Update "use PSA" documentation (inc. strategy)

### LINKED ISSUE #5621: Update "use PSA" documentation
### Suggested enhancement

This documents the effects of the config option `MBEDTLS_USE_PSA_CRYPTO` in detail. It is not updated in each PR that extends `USE_PSA_CRYPTO` because that would be an endless source of conflicts. However, it should be updated before releasing.

Note: if we have achieved most of the "part 2" EPIC before the next release, we'll probably want to switch the description from "does A, B and C" to "does everything except for X, Y and Z".

Edit: all documents in `docs/architecture/psa-migration` need to be reviewed to make sure they're up to date.

### Justification

Mbed TLS needs this because out-of-date documentation is bad.

### PR DESCRIPTION
**Pull request by mpg**
## Description

Fixes #5621

## Status

**IN DEVELOPMENT** - but what's there can already be reviewed. Will be finalized when we stabilize in preparation for 3.2.

Should be fully up to date as of 271c3052fb8c46819faaa42bb1efb29ea3129eeb - just anticipates a bit on #5831 and #5844 being closed before the release.

Possible changes that may or may not happen before the release, and will require an update if they do:

- EC J-PAKE #4153 -> `psa-limitations.md`, #5847 -> `use-psa-crypto.md`
- RSA-PSS? depending on #5277 and possible follow-up work -> `psa-limitations.md` + `use-psa-crypto.md`

## Requires Backporting

NO  - 3.x only

### CODE REVIEW

File: docs/architecture/psa-migration/psa-limitations.md [hunk 1]
```diff
@@ -345,40 +354,6 @@ https://github.com/ARM-software/psa-crypto-api/issues/539
...
-
-Note: if later we want to also isolate short-term secret (G3), the "secondary
-secret" (output of DHE/ECDHE key agreement or RSA decryption) could be a
-candidate. This wouldn't be a problem as the PSA key derivation API always
-allows inputs from key slots. (Tangent: the hard part in isolating the result
-of RSA decryption would be still checking that is has the correct format:
-48 bytes, the first two matching the TLS version - note that this is timing
-sensitive.)
-
 HKDF: Expand not exposed on its own (TLS 1.3)
```
Comment (gilles-peskine-arm) [inline docs/architecture/psa-migration/psa-limitations.md:357]: https://github.com/Mbed-TLS/mbedtls/issues/5784

**Comment by mpg** [comment]
I updated the second paragraph of ["PSA implementation in Mbed TLS" in the README](https://github.com/Mbed-TLS/mbedtls#psa-implementation-in-mbed-tls) but the third paragraph needs updating too (outdated link), and perhaps the first as well (do we still consider that PSA code is not as well reviewed as the rest?). @gilles-peskine-arm could you suggest a new version? Then I'd include it in this PR, to avoid the trouble of an extra PR with contextual conflicts.

**Comment by gilles-peskine-arm** [comment]
First paragraph: I think we can drop the maturity bit. We haven't done the systematic quality bar that I wanted to do, but we've rewritten pretty much all the original code and test coverage, so I think we've reached similar quality to the old code.

Third paragraph: we've passed compliance with 1.0. We can just remove this paragraph. Although we might tweak the first paragraph to say that while we're a reference implementation, we don't aim to implement the whole thing, in particular we don't implement all the algorithms.

**Comment by mpg** [comment]
Self-reminder: add ChangeLog entries too.

**Review by minosgalanakis — DISMISSED** [review]
LGTM

**Review by wernerlewis — DISMISSED** [review]
LGTM

**Comment by mpg** [comment]
There was a conflict in `mbedtls_config.h` so I rebased and force-pushed in order to resolve it. @minosgalanakis @wernerlewis please review. (I took the opportunity to mention the running handshake hash as shared between 1.2 and 1.3, which I had forgotten before - other than that there should be not change.)

**Review by wernerlewis — APPROVED** [review]

**Pull request merged**